### PR TITLE
Fix GenotypeConcordance bug writing spanning deletion

### PR DESCRIPTION
**Pull request by kachulis**
### Description

When running ``GenotypeConcordance`` and writing out an annotated vcf, picard will currently crash if there is a het snp with overlapping deletion in one vcf, and an indel called at the same location in the other vcf.  For example:

EVAL:
chr 1  18359383  A        G,\*            1|2

TRUTH:
chr 1  18359383 AAAGGAAGG      A       0,1

when combining these calls into a single line in the output vcf, the snp call must have the deleted reference bases of the indel appended to it so that the snp *A->G* becomes *AAAGGAAGG -> GAAGGAAGG*.  Currently the deleted reference bases will in this case also be appended to the *\** allele, leading to an attempt to write out an alt allele *\*AAGGAAGG*, which causes a crash in htsjdk.  This PR fixes this issue by explicitly never appending to the *\** allele in ``GenotypeConcordance.spliceOrAppendString``.

----

### Checklist (never delete this)

Never delete this, it is our record that procedure was followed. If you find that for whatever reason one of the checklist points doesn't apply to your PR, you can leave it unchecked but please add an explanation below.

#### Content
- [x] Added or modified tests to cover changes and any new functionality
- [ ] Edited the README / documentation (if applicable)
- [ ] All tests passing on Travis

#### Review
- [ ] Final thumbs-up from reviewer
- [ ] Rebase, squash and reword as applicable

For more detailed guidelines, see https://github.com/broadinstitute/picard/wiki/Guidelines-for-pull-requests

### CODE REVIEW

File: src/test/java/picard/vcf/GenotypeConcordanceTest.java [hunk 1]
```diff
@@ -120,7 +125,8 @@ public void tearDown() {
                 {CEU_TRIOS_SNPS_VCF, "NA12878", CEU_TRIOS_SNPS_VCF, "NA12878", null, null, true, false, CEU_TRIOS_SNPS_VS_CEU_TRIOS_SNPS_GC_ALL_ROWS},
                 {CEU_TRIOS_SNPS_VCF, "NA12878", CEU_TRIOS_SNPS_VCF, "NA12891", 40, null, false, false, CEU_TRIOS_SNPS_VS_CEU_TRIOS_SNPS_GC_MIN_GQ},
                 {CEU_TRIOS_SNPS_VCF, "NA12878", CEU_TRIOS_SNPS_VCF, "NA12891", null, 40, false, false, CEU_TRIOS_SNPS_VS_CEU_TRIOS_SNPS_GC_MIN_DP},
-                {NIST_MISSING_SITES_TRUTH_VCF, "NA12878", CEU_TRIOS_SNPS_VCF, "NA12878", null, null, false, true, NIST_TRUTH_SNPS_VS_CEU_TRIOS_SNPS_GC}
+                {NIST_MISSING_SITES_TRUTH_VCF, "NA12878", CEU_TRIOS_SNPS_VCF, "NA12878", null, null, false, true, NIST_TRUTH_SNPS_VS_CEU_TRIOS_SNPS_GC},
+                {SPANNING_DELETION_TRUTH.toFile(),"/dev/stdin",SPANNING_DELETION_CALLSET.toFile(),"CHMI_CHMI3_WGS2",null,null,false,false,SPANNING_DELETION_CALLSET_VS_SPANNING_DELETION_TRUTH}
```
Comment (yfarjoun) [inline src/test/java/picard/vcf/GenotypeConcordanceTest.java]: spaces after commas

File: src/main/java/picard/vcf/GenotypeConcordance.java [hunk 2]
```diff
@@ -746,16 +746,13 @@ public Alleles(final OrderedSet<String> allAlleles,
      * destination string, the given string is appended to the destination.
```
Comment (yfarjoun) [inline src/main/java/picard/vcf/GenotypeConcordance.java]: modify the javaDoc

**Comment by yfarjoun** [comment]
Also note that there was an attempt to fix this issue here: https://github.com/broadinstitute/picard/pull/904

File: src/test/java/picard/vcf/GenotypeConcordanceTest.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (kachulis) [reply src/test/java/picard/vcf/GenotypeConcordanceTest.java]: this is the sample name in */testdata/picard/vcf/spanningDeletionTruth.vcf*.  why it is given that name in the original vcf I do not know.

**Pull request merged**